src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/config/WebClientConfiguration.kt

```diff
@@ -119,8 +118,14 @@ class WebClientConfiguration(
         retryTemplate.setListeners(arrayOf(RetryLogger("token request failed; retrying")))
 
         retryTemplate.execute(
-          RetryCallback<ClientHttpResponse, RestClientException> {
-            execution.execute(request, body)
+          RetryCallback {
+            execution.execute(request, body).also {
+              // i hate that this basically duplicates `RestTemplate.handleResponse`,
+              // but it's the only way i could figure out to get this to work.
```

Comment:
_not a blocker_
could listeners also work?

Reply:
you can't alter the request processing with listeners, only run blocks of code when certain things happen in the flow